data/resources/ui/content-message-media.ui

```diff
@@ -0,0 +1,49 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<interface>
+  <template class="ContentMessageMedia" parent="GtkWidget">
+    <style>
+      <class name="message-bubble"/>
+      <class name="message-media"/>
+    </style>
+    <child>
+      <object class="GtkBox" id="content">
```

Comment:
I wonder whether this indirection of having a `GtkBox` as a single child is necessary.
Could we not set the layout-manger like in the attached patch (just remove the ".zip" ending)?

[media-no-indirection.patch.zip](https://github.com/melix99/telegrand/files/7818240/media-no-indirection.patch.zip)

Reply:
Nope, setting a layout-manager will basically overwrite the manual measure implementation.

I'll explain a bit of the why this widget is implemented like that. Basically I want this widget to be a picture + caption in a vertical box, but with a small difference than a normal GtkBox: the width of the box must be only calculated by the picture. This means that the caption must not require more width than what the picture wants.

To do this _properly_ we'd probably need to create ours custom GtkBoxLayout, but that's not easy at all as it may seem! I could explain the details to implement such thing, like for example that we'd need a binary search to correctly calculate the needed width of this widget, which is something that GtkBoxLayout already does. So I decided to create a custom GtkWidget that tells to its single child (the GtkBox) that its width must be calculated based on the picture. This is a bit of a workaround, but this is the simplest way to get this working decently.